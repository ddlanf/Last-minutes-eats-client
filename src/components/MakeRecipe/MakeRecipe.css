/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type=number] {
  -moz-appearance: textfield;
}

.make-recipe-preparation-time-input:focus,
.make-recipe-preparation-time-unit-input:focus,
.make-recipe-variable-input:focus,
.make-recipe-input:focus,
.make-recipe-minus:focus,
.make-recipe-plus:focus{
    outline: none;
}

.make-recipe{
    width: 80%;
    max-width: 1300px;
    margin-left: auto;
    margin-right: auto;
    margin-top: 20%;
    text-align: center;
}

.make-recipe-heading{
    font-size: 5vw;
    margin-top: 2%;
}

.make-recipe-form{
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 10%;
}

.make-recipe-label{
    display: block;
    font-size: 4.5vw;
    margin-top: 1vw;
    margin-bottom: 2%;
}

.make-recipe-input{
    width: 90%;
    margin-bottom: 5%;
    font-size: 4.5vw;
    max-width: 500px;
    border: none;
    border-bottom: 1px solid black;
}

.make-recipe-instructions-input-box{
    height: 50vw;
    width: 100%;
    font-size: 3vw;
    padding-top: 4%;
    padding-bottom: 8%;
    overflow-y: auto;
    margin-left: auto;
    margin-right: auto;
}

.make-recipe-ingredients-input-box{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    font-size: 3vw;
    overflow: hidden;
}

.make-recipe-button-and-text{
    width: 100%;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

input.make-recipe-variable-input{
    width: 80%;
    font-size: 4.5vw;
    border: 1px solid black;
}

textarea{
    resize: none;
    overflow:hidden;
}

textarea.make-recipe-variable-input{
    width: 80%;
    font-size: 4.5vw;
    height: 22vw;
    border: 1px solid black;
}

.make-recipe-number{
    position: absolute;
    bottom: 68%;
    left: 5%;
    font-size: 5vw;
}

.make-recipe-plus{
    position: absolute;
    bottom: 0;
    right: 0;
    border-radius: 100%;
    height: 6vw;
    width: 6vw;
    font-size: 5vw;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border: none;
    color: white;
    background-color:#006691;
}

.make-recipe-plus:active, .make-recipe-minus:active{
    color: black;
}

.make-recipe-minus{
    position: absolute;
    left: 0;
    bottom: 0;
    height: 6vw;
    width: 6vw;
    font-size: 5vw;
    border-radius: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: rgb(197, 37, 37);
    border: none;
    color: white;
}

.make-recipe-preparation-time-label{
    display: block;
    margin-top: 6vw;
    font-size: 4vw;
}

.make-recipe-preparation-time-box{
    margin-top: 3%;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-end;
}

.make-recipe-preparation-time-label{
    font-size: 5vw;
}

.make-recipe-preparation-time-input{
    font-size: 5vw;
    width: 10%;
    border: none;
    border-bottom: 1px solid black;
}

.make-recipe-preparation-time-unit-input{
    font-size: 5vw;
    border: none;
    border-bottom: 1px solid black;
}

.make-recipe-error, .make-recipe-buffer{
    font-size: 4.5vw;
    color: red;
}

.make-recipe-submit, .make-recipe-back{
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 10%;
    padding: 2%;
    font-size: 6vw;
}

.make-recipe-submit{
    background-color: #209463;
    border: none;
    border-radius: 3rem;
    padding: .5rem 2rem;
    color: rgb(42, 48, 71);
}

.make-recipe-back{
    background-color: #696969;
    border: none;
    border-radius: 2.6rem;
    padding: .5rem 1.5rem;
    color: rgb(37, 43, 68);
    
}


@media screen and (min-width: 1000px) {

    .make-recipe{
        width: 100%;
        margin-top: 10%;
    }

    .make-recipe-top-field{
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
    }

    .make-recipe-label{
        font-size: 3vw;
        margin-top: 1vw;
    }

    .instructions.make-recipe-label{
        margin-bottom: 0;
    }

    .make-recipe-instructions-input-box{
        height: 22vw;
    }

    .make-recipe-ingredients-and-instructions{
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: flex-start;
        width: 100%;
    }

    .make-recipe-button-and-text{
        width: 100%;
    }

    .make-recipe-minus, .make-recipe-plus{
        height: 45px;
        width:  45px;
        font-size: 40px;
    }

    .make-recipe-number{
        font-size: 40px;
    }
  
    .make-recipe-input{
        font-size: 3vw;
    }

    .make-recipe-variable-input{
        font-size: 3vw;
    }

    input.make-recipe-variable-input{
        width: 80%;
        font-size: 4.5vw;
        border: 1px solid black;
    }
    
    .make-recipe-ingredients-field{
        width: 45%;
    }

    .make-recipe-instructions-field{
        width: 55%;
    }

    input.make-recipe-variable-input{
        font-size: 3vw;
        margin: 0;
    }

    textarea.make-recipe-variable-input{
        font-size: 3vw;
        width: 80%;
        height: 10vw;
    }

    .make-recipe-error, .make-recipe-buffer{
        font-size: 3vw;
        color: red;
    }

    .make-recipe-preparation-time-label{
        display: block;
        margin-top: 2%;
        font-size: 3vw;
    }

    .make-recipe-preparation-time-unit-input, .make-recipe-preparation-time-input{
        font-size: 3vw;
    }

    .make-recipe-preparation-time-input{
        width: 5%;
    }

    .make-recipe-submit, .make-recipe-back{
        margin-top: 5vw;
        font-size: 3vw;
    }
}