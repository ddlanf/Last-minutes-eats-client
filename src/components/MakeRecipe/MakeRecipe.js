import React, { Component } from 'react'
import './MakeRecipe.css'

export default class MakeRecipe extends Component {

    constructor(props){
        super(props)
        this.state = {
            recipe_name: '',
            Image:'',
            preparation_time: null,
            preparation_time_unit: 'minutes',
            ingredients: [
                { value: '', display: true  },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
            ],
            steps : [
                { value: '', display: true  },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
                { value: '', display: false },
            ]
        }
    }

    goBack = () =>{
        this.props.history.push(`/view-all-recipes`)
    }

    handleInputChange = (event) =>{

        this.setState({ clicked: false })

        const { name } = event.target
        const { value } = event.target
        
        this.setState({
             [name] : value 
        })

    }

    handleArrayInputChange = (event, input, index) =>{
        const currentInputs = this.state[input]
        currentInputs[index].value = event.target.value
        
        this.setState({
            input: currentInputs
        })
    }   

    handleSelectChange = (event) =>{
        this.setState({ preparation_time_unit : event.value});
    }

    makeIngredientslist(){
        const ingredientsInput = [1,2,3,4,5].map((num, index)=>{
            return( 
                <div 
                    key={num} 
                    style={{display: this.state.ingredients[index].display ? 'block' : 'none' }}
                    className="make-recipe-button-and-text">
                    {(index !== 4) ?
                        <button 
                            style={{display: 
                                this.state.ingredients[index + 1].display ? 'none' :  'block'}}
                            onClick={() => this.addAnother('ingredients', index)}
                            className="make-recipe-plus">
                            +
                        </button>
                        : ''}
                    <input 
                        className="make-recipe-variable-input" 
                        type="text"
                        name="ingredient-1"
                        onChange={(e) => this.handleArrayInputChange(e, 'ingredients', index)}/>
                     {(index !== 0) ?
                        <button 
                            type="button"
                            style={{display: (index === 4) ? 'block' : this.state.ingredients[index + 1].display  
                                ? 'none' :  'block'}}
                            onClick={() => this.deleteOne('ingredients', index)}
                            className="make-recipe-minus">
                            -
                        </button>
                        : ''}
                </div>
            )                          
        })

        return ingredientsInput
    }

    makeInstructionlists(){
         const stepsInput = [1,2,3,4,5,6,7].map((num, index)=>{
             return (
                <div 
                    key={num} 
                    style={{display: this.state.steps[index].display ? 'block' : 'none' }}
                    className="make-recipe-button-and-text">
                     {(index !== 6) ?
                        <button 
                            type="button"
                            style={{display: 
                                this.state.steps[index + 1].display ? 'none' :  'block'}}
                            onClick={() => this.addAnother('steps', index)}
                            className="make-recipe-plus">
                            +
                        </button>
                        : ''}
                    <input 
                        className="make-recipe-variable-input" 
                        type="text"
                        name={`step-${num}`}
                        onChange={(e) => this.handleArrayInputChange(e, 'steps', index)}/>
                     {(index !== 0) ?
                        <button 
                            type="button"
                            style={{display: (index === 6) ? 'block' : this.state.steps[index + 1].display  
                                ? 'none' :  'block'}}
                            onClick={() => this.deleteOne('steps', index)}
                            className="make-recipe-minus">
                            -
                        </button>
                        : ''}
                </div>
             )
         })

         return stepsInput
    }   

    addAnother = (input, index) =>{
        const currentInputs = this.state[input]
        currentInputs[index + 1].display = true
        
        this.setState({
            input: currentInputs
        })
    }

    deleteOne = (input, index) =>{
        const currentInputs = this.state[input]
        currentInputs[index].display = false

        this.setState({
            input: currentInputs
        })
    }


    postRecipe = (e) =>{
        e.preventDefault()

        const { recipe_name, image, preparation_time, preparation_time_unit } = this.state
        const ingredients = this.state.ingredients.filter(item => item.display).map(item => item.value)
        console.log(ingredients)
    }

    render() {
        return (
            <>
                <section className="make-recipe">
                    <h1 className="make-recipe-heading">Make Your Recipe</h1>
                    <form className="make-recipe-form"
                        onSubmit={(e) => this.postRecipe(e)}
                        >
                        <label className="make-recipe-label">Recipe Name</label>
                        <input 
                            className="make-recipe-input" 
                            type="text"
                            name="recipe_name"
                            onChange={this.handleInputChange}/>
                        <label className="make-recipe-label">Recipe Image Link</label>
                        <input 
                            className="make-recipe-input" 
                            type="text"
                            onChange={this.handleInputChange}/>
                        <div className="make-recipe-ingredients-and-steps">
                                <div className="make-recipe-ingredients-input">
                                    <label className="make-recipe-label">Ingredients</label> 
                                    {this.makeIngredientslist()}
                                </div>
                                <div className="make-recipe-instructions-input">
                                    <label className="make-recipe-label">Instructions</label>
                                    {this.makeInstructionlists()}
                                </div>
                        </div>
                        <label className="make-recipe-preparation-time-label" >Preparation Time</label>
                        <div className="make-recipe-preparation-time-box">
                            <input
                                className="make-recipe-preparation-time-input" 
                                type="number"
                                name="preparation_time"
                                onChange={this.handleInputChange}
                            />
                            <select 
                                className="make-recipe-preparation-time-unit-input" 
                                value={this.state.preparation_time_unit} onChange={(e) => this.handleSelectChange(e.target)}>
                                <option value="seconds">sec</option>
                                <option value="minutes">min</option>
                            </select>
                        </div>
                        <button
                            type="submit" 
                            className="make-recipe-submit">
                            Submit
                        </button>
                        <button 
                            type="button"
                            onClick={this.goBack}
                            className="edit-recipe-submit"
                            >
                            Back
                        </button>
                    </form>
                </section>   
            </>
        )
    }
}
