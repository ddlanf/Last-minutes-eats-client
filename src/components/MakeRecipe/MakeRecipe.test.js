import React from 'react';
import ReactDOM from 'react-dom';
import MakeRecipe from './MakeRecipe';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/index'

describe('Make component', () => {
    
    const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>   
                      <Provider store={store}>
                        <MakeRecipe/>
                      </Provider>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>   
                  <Provider store={store}>
                    <MakeRecipe/>
                  </Provider>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});