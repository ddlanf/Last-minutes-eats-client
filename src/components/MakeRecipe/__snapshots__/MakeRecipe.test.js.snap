// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make component renders the UI as expected 1`] = `
<section
  className="make-recipe"
>
  <h1
    className="make-recipe-heading"
  >
    Make Your Recipe
  </h1>
  <form
    className="make-recipe-form"
    onSubmit={[Function]}
  >
    <div
      className="make-recipe-top-field"
    >
      <div
        className="make-recipe-name-field"
      >
        <label
          className="make-recipe-label"
        >
          Recipe Name
        </label>
        <input
          className="make-recipe-input"
          name="recipe_name"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="make-recipe-image-field"
      >
        <label
          className="make-recipe-label"
        >
          Recipe Image Link
        </label>
        <input
          className="make-recipe-input"
          name="image"
          onChange={[Function]}
          type="text"
        />
      </div>
    </div>
    <div
      className="make-recipe-ingredients-and-instructions"
    >
      <div
        className="make-recipe-instructions-field"
      >
        <label
          className="make-recipe-label instructions"
        >
          Instructions
        </label>
        <div
          className="make-recipe-instructions-input-box"
        >
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "block",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              1
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-1"
              onChange={[Function]}
              type="text"
            />
            
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              2
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-2"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              3
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-3"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              4
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-4"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              5
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-5"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              6
            </span>
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <textarea
              className="make-recipe-variable-input"
              name="step-6"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <span
              className="make-recipe-number"
            >
              7
            </span>
            
            <textarea
              className="make-recipe-variable-input"
              name="step-7"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
        </div>
      </div>
      <div
        className="make-recipe-ingredients-field"
      >
        <label
          className="make-recipe-label"
        >
          Ingredients
        </label>
        <div
          className="make-recipe-ingredients-input-box"
        >
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "block",
              }
            }
          >
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <input
              className="make-recipe-variable-input"
              name="ingredient-1"
              onChange={[Function]}
              type="text"
            />
            
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <input
              className="make-recipe-variable-input"
              name="ingredient-1"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <input
              className="make-recipe-variable-input"
              name="ingredient-1"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <button
              className="make-recipe-plus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              +
            </button>
            <input
              className="make-recipe-variable-input"
              name="ingredient-1"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
          <div
            className="make-recipe-button-and-text"
            style={
              Object {
                "display": "none",
              }
            }
          >
            
            <input
              className="make-recipe-variable-input"
              name="ingredient-1"
              onChange={[Function]}
              type="text"
            />
            <button
              className="make-recipe-minus"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              -
            </button>
          </div>
        </div>
      </div>
    </div>
    <label
      className="make-recipe-preparation-time-label"
    >
      Preparation Time
    </label>
    <div
      className="make-recipe-preparation-time-box"
    >
      <input
        className="make-recipe-preparation-time-input"
        name="preparation_time"
        onChange={[Function]}
        type="number"
      />
      <select
        className="make-recipe-preparation-time-unit-input"
        onChange={[Function]}
        value="minutes"
      >
        <option
          value="seconds"
        >
          sec
        </option>
        <option
          value="minutes"
        >
          min
        </option>
      </select>
    </div>
    <div
      className="make-recipe-token-box"
    >
      <label
        className="make-recipe-token-label"
      >
        Create a Token
        <span
          className="make-recipe-break"
        >
          (This will be required to edit or delete your recipe)
        </span>
      </label>
      <input
        className="make-recipe-token-input"
        name="token"
        onChange={[Function]}
        type="password"
      />
    </div>
    
    <button
      className="make-recipe-submit"
      disabled={true}
      name="submit"
      style={
        Object {
          "opacity": "0.5",
        }
      }
      type="submit"
    >
      Submit
    </button>
    <button
      className="make-recipe-back"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</section>
`;
